src/lib/device-types/storage/gh.ts

```diff
@@ -0,0 +1,46 @@
+import * as Octokit from '@octokit/rest';
+import * as _ from 'lodash';
+import * as Promise from 'bluebird';
+
+import {
+	DEVICE_TYPE_REPO_NAME,
+	DEVICE_TYPE_REPO_OWNER,
+	DEVICE_TYPE_REPO_PATH,
+	DEVICE_TYPE_REPO_REF,
+} from '../../config';
+
+interface FetchDeviceTypeOptions {
+	owner: string;
+	repo: string;
+	path: string;
+	ref?: string;
+}
+
+export const fetchExternalDeviceTypes = (
+	options?: FetchDeviceTypeOptions,
+): Promise<string[]> => {
+	const realOptions = options || {
+		owner: DEVICE_TYPE_REPO_OWNER,
+		repo: DEVICE_TYPE_REPO_NAME,
+		path: DEVICE_TYPE_REPO_PATH,
+		ref: DEVICE_TYPE_REPO_REF,
+	};
+	const octokit = new Octokit();
+	const dts: string[] = [];
+	return Promise.try(() => {
+		return octokit.repos.getContents(realOptions).then(dtSlugs => {
+			if (_.isArray(dtSlugs.data)) {
+				dtSlugs.data.forEach(element => {
+					if (element.type == 'dir') {
+						dts.push(element.name);
+					}
+				});
```

Comment:
Could do this in a filter?

```suggestion
    dts = dtSlugs.data.filter(e => e.type == 'dir').map(e => e.name);
```